bokehjs/src/lib/models/glyphs/glyph.ts

```diff
@@ -179,21 +178,40 @@ export abstract class GlyphView extends View {
 
   draw_legend_for_index(_ctx: Context2d, _bbox: Rect, _index: number): void {}
 
-  hit_test(geometry: Geometry): hittest.HitTestResult {
-    let result = null
+  protected _hit_point?(geometry: geometry.PointGeometry): Selection
+  protected _hit_span?(geometry: geometry.SpanGeometry): Selection
+  protected _hit_rect?(geometry: geometry.RectGeometry): Selection
+  protected _hit_poly?(geometry: geometry.PolyGeometry): Selection
+
+  hit_test(geometry: geometry.Geometry): hittest.HitTestResult {
+    switch (geometry.type) {
+      case "point":
+        if (this._hit_point != null)
+          return this._hit_point(geometry)
+        break
+      case "span":
+        if (this._hit_span != null)
+          return this._hit_span(geometry)
+        break
+      case "rect":
+        if (this._hit_rect != null)
+          return this._hit_rect(geometry)
+        break
+      case "poly":
+        if (this._hit_poly != null)
+          return this._hit_poly(geometry)
+        break
+    }
 
-    const func = `_hit_${geometry.type}`
-    if ((this as any)[func] != null) {
-      result = (this as any)[func](geometry)
-    } else if (this._nohit_warned[geometry.type] == null) {
+    if (this._nohit_warned.has(geometry.type)) {
```

Comment:
Is this missing a negation? We want to warn if the type *hasn't* been added

Reply:
Indeed. I added some tests this time.